src/doubly_link.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+"""."""
+
+
+class Node(object):
+    """Docstring for Node."""
+
+    def __init__(self, data=None, previous_node=None, next_node=None):
+        """Initializer for the class instance."""
+        self.data = data
+        self.next_node = next_node
+        self.previous_node = previous_node
+
+
+class DoublyLink(object):
+    """Docstring for DoublyLink."""
+
+    def __init__(self, data=None):
+        """Initialize class instance."""
+        self.head = None
+        self.tail = None
+        self._length = 0
+
+        if type(data) in [list, tuple, str]:
+            for item in data:
+                self._length += 1
+                self.push(item)
+
+        elif data is not None:
+            raise TypeError('Requires an iterable value.')
+
+    def push(self, val):
+        """Insert the value 'val' at the head of the list."""
+        if not val:
+            raise ValueError('You must provide a not-null value.')
+        if self.head is None:
+            self.head = Node(val)
+            self.tail = self.head
+        elif self._length >= 1:
+            new_node = Node(val)
+            current = self.head
+            self.head = new_node
+            self.head.next_node = current
+            self.head.next_node.previous_node = self.head
+        self._length += 1
+        return self
+
+    def size(self):
+        """Will return the length of the list."""
+        return self._length
+
+    def append(self, val):
+        """Append val at tail."""
+        new_node = Node(val)
+        if self.head is None:
+            self.head = new_node
+            self.tail = new_node
+            self.next_node = new_node
+        else:
+            self.tail.next_node = new_node
+            new_node.previous_node = self.tail
+            self.tail = new_node
+            self._length += 1
+        return self
+
+    def pop(self):
+        """Pop head, raise exception."""
```

Comment:
This is going to pop and also raise the exception? Is there a condition for raising the exception?